### PR TITLE
[WIP] Healing remove stale objects

### PR DESCRIPTION
**Pull request by vadmeste**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

File: cmd/xl-v1-healing.go [hunk 1]
```diff
@@ -679,22 +685,31 @@ func (xl xlObjects) HealObject(ctx context.Context, bucket, object string, dryRu
...
 	// Read metadata files from all the disks
 	partsMetadata, errs := readAllXLMetadata(healCtx, storageDisks, bucket, object)
 
 	latestXLMeta, err := getLatestXLMeta(healCtx, partsMetadata, errs)
 	if err != nil {
+		if err == errFileNotFound {
+			// Check if there some xl.json found in some disks, move them to trash
+			for i, disk := range storageDisks {
+				if errs[i] == nil {
+					disk.DeleteFile(minioMetaTrashBucket, object)
```
Comment (harshavardhana) [inline cmd/xl-v1-healing.go]: we should do this for missing `part.1`s as well

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by vadmeste** [comment]
Fixing this https://github.com/minio/minio/issues/6575 will make an object visible even if it doesn't have read quorum, which means we can remove it using regular `mc rm` command.

Downloading it will generate an error anyway.

**Comment by harshavardhana** [comment]
This is not considered as a valid fix for this issue after discussing with @abperiasamy, the basic fundamental is healing doesn't remove any data, https://github.com/minio/minio/pull/6592 has enough requirements.

This PR can be closed.

**Comment by vadmeste** [comment]
> This is not considered as a valid fix for this issue after discussing with @abperiasamy, the basic fundamental is healing doesn't remove any data, #6592 has enough requirements.

Yes, closing

**Pull request closed without merging**